### PR TITLE
Added Drivetrain data logging

### PR DESCRIPTION
**Pull request by KinuTheDragon**
# Purpose: Added drivetrain data logging

# Changes:
- build.gradle: Added AdvantageKit library
- Constants.java: Added logger constants
- Robot.java: Added code to set up logger
- RobotContainer.java and SequentialAutoCommand.java: Replaced deprecated methods with non-deprecated equivalents
- Drivetrain.java: Start logging values that were originally only on SmartDashboard
- vendordeps: Updated libraries

### CODE REVIEW

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -27,7 +37,7 @@
  * creating this project, you must also update the build.properties file in 
  * the project.
  */
-public class Robot extends TimedRobot {
+public class Robot extends LoggedRobot {
```
Comment (bjmcternan) [inline src/main/java/frc/robot/Robot.java:40]: Does LoggedRobot extend TimedRobot?

Comment (bjmcternan) [reply src/main/java/frc/robot/Robot.java:40]: We should be careful that we're not missing anything by changing this inheritance

**Review by bjmcternan — APPROVED** [review]
Looks good. Nice work. Let's just answer the one question I had and then you can merge this

File: src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (KinuTheDragon) [reply src/main/java/frc/robot/Robot.java:40]: LoggedRobot does not extend TimedRobot. Both of them implement IterativeRobotBase. However, from [the AdvantageKit installation docs](https://github.com/Mechanical-Advantage/AdvantageKit/blob/main/docs/INSTALLATION.md):
> `LoggedRobot` performs the same functions as `TimedRobot`, with some exceptions:
> - It does not support adding extra periodic functions.
> - The method `setUseTiming` allows the user code to disable periodic timing and run cycles as fast as possible during replay. The timestamp read by methods like `Timer.getFPGATimstamp()` will still match the timestamp from the real robot.
I'll check to ensure that both of these still work with our code.

Comment (bjmcternan) [reply src/main/java/frc/robot/Robot.java:40]: OK that probably won't be an issue

Comment (bjmcternan) [reply src/main/java/frc/robot/Robot.java:40]: You'll need the others to approve before you can merge

**Review by GoldCrafter — APPROVED** [review]
Looks good.

File: src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (KinuTheDragon) [reply src/main/java/frc/robot/Robot.java:40]: I just checked. Changing it over will not cause errors.

**Pull request merged**